cmd/yorkie/version.go

```diff
@@ -17,27 +17,152 @@
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
 
+	"github.com/yorkie-team/yorkie/admin"
+	"github.com/yorkie-team/yorkie/api/types"
+	"github.com/yorkie-team/yorkie/cmd/yorkie/config"
 	"github.com/yorkie-team/yorkie/internal/version"
 )
 
+var (
+	clientOnly bool
+	output     string
+)
+
 func newVersionCmd() *cobra.Command {
 	return &cobra.Command{
-		Use:   "version",
-		Short: "Print the version number of Yorkie",
+		Use:     "version",
+		Short:   "Print the version number of Yorkie",
+		PreRunE: config.Preload,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			fmt.Printf("Yorkie: %s\n", version.Version)
-			fmt.Printf("Go: %s\n", runtime.Version())
-			fmt.Printf("Build date: %s\n", version.BuildDate)
+			if err := Validate(); err != nil {
+				return err
+			}
+
+			var versionInfo types.VersionInfo
+			versionInfo.ClientVersion = getYorkieClientVersion()
+
+			serverVersionChan := make(chan *types.VersionDetail)
+			errorChan := make(chan error)
+
+			if !clientOnly {
+				go func() {
+					rpcAddr := viper.GetString("rpcAddr")
+					auth, err := config.LoadAuth(rpcAddr)
+					if err != nil {
+						errorChan <- err
+						return
+					}
+
+					cli, err := admin.Dial(rpcAddr, admin.WithToken(auth.Token), admin.WithInsecure(auth.Insecure))
+					if err != nil {
+						errorChan <- err
+						return
+					}
+					defer cli.Close()
+
+					ctx := context.Background()
+					sv, err := cli.GetServerVersion(ctx)
+					if err != nil {
+						errorChan <- err
+						return
+					}
+
+					serverVersionChan <- sv
+				}()
+			}
+
+			var serverErr error
+
+			if !clientOnly {
+				select {
+				case sv := <-serverVersionChan:
+					versionInfo.ServerVersion = sv
+				case err := <-errorChan:
+					serverErr = err
+				}
+			}
+
+			switch output {
+			case "":
+				cmd.Printf("Yorkie Client: %s\n", versionInfo.ClientVersion.YorkieVersion)
+				cmd.Printf("Go: %s\n", versionInfo.ClientVersion.GoVersion)
+				cmd.Printf("Build Date: %s\n", versionInfo.ClientVersion.BuildDate)
+				if versionInfo.ServerVersion != nil {
+					cmd.Printf("Yorkie Server: %s\n", versionInfo.ServerVersion.YorkieVersion)
+					cmd.Printf("Go: %s\n", versionInfo.ServerVersion.GoVersion)
+					cmd.Printf("Build Date: %s\n", versionInfo.ServerVersion.BuildDate)
+				}
+			case "yaml":
+				marshalled, err := yaml.Marshal(&versionInfo)
+				if err != nil {
+					return errors.New("failed to marshal YAML")
+				}
+				fmt.Println(string(marshalled))
+			case "json":
+				marshalled, err := json.MarshalIndent(&versionInfo, "", "  ")
+				if err != nil {
+					return errors.New("failed to marshal JSON")
+				}
+				fmt.Println(string(marshalled))
+			}
+
+			if serverErr != nil {
+				cmd.Printf("Error fetching server version: ")
+				if strings.Contains(serverErr.Error(), "unimplemented") {
+					cmd.Printf("The server does not support this operation. You might need to check your server version.\n")
+				} else {
+					cmd.Print(serverErr)
+				}
+			}
+
```

Comment:
When calling the version command, the grpc response code received from the server was unknown, which led to difficulties in error filtering.
So, I added it because I thought a detailed message would be most needed if the server does not support the version command.
:confused: This part also looks bad, judging by the `unimplemented` keyword in the error message. Isn't there a better way?

Reply:
@hyun98 How about following the process or error handling in other CLI commands like `login.go`.
Since this PR is about adding server version check features to CLI, lets follow the conventions for now.
After this PR is merged, we can talk about the current error handling of CLI commands and improve that.